sdk/cosmos/azure_data_cosmos/src/clients/cosmos_client.rs

```diff
@@ -198,17 +228,18 @@ impl CosmosClient {
             id: &'a str,
         }
 
-        let url = self.pipeline.url(&self.databases_link);
-        let mut req = Request::new(url, Method::Post);
-        req.insert_headers(&options.throughput)?;
-        req.insert_headers(&ContentType::APPLICATION_JSON)?;
-        req.set_json(&RequestBody { id })?;
+        let body = serde_json::to_vec(&RequestBody { id })?;
+        let builder = CosmosRequestBuilder::new(OperationType::Create, ResourceType::Databases);
+        let cosmos_request = builder
+            .headers(&options.throughput)
+            .body(Some(body))
```

Comment:
nit: I think it would be good to have a `.json<T: Serialize>(mut self, value: T) -> Self` method on `CosmosRequestBuilder` that handles the serialization and setting of the body.

Reply:
Yes - thanks for the suggestion. This has been refactored.